src/util/sfdcUrl.ts

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { URL } from 'url';
+import { Env, Duration } from '@salesforce/kit';
+import { MyDomainResolver } from '../status/myDomainResolver';
+
+export class SfdcUrl extends URL {
+  /**
+   * Salesforce URLs.
+   */
+  public static SANDBOX = 'https://test.salesforce.com';
+  public static PRODUCTION = 'https://login.salesforce.com';
+
+  public constructor(input: string, base?: string | URL) {
+    super(input, base);
+    if (this.protocol !== 'https:') {
```

Comment:
@maggiben Thomas had mentioned trying to limit the number of warnings emitted to just one per command execution, just to keep the noise to user down. We should consider finding a way to accomplish this. Possibly a cache of some sort with the key being the origin.

Reply:
I decided to keep this branch "clean" (only the class and it's tests) until everyone feels happy about the functionality and then move on to refactor phase.

Reply:
2 PRs seems fine.  It's good to add a note to the PR description if you're planning to do that, so the reviewers understand your intent.

Reply:
A cache is a good idea.  As more things use this class, getting lots of http warnings on the same command would be painful.
A list of urls that have already been warned about (and maybe a timestamp?) could be a nice touch.  Since it would only get executed when people are doing something insecure, it would never hit most users with any perf problems.  

This is mostly for internals, anyway I think.  

Maybe we can do it via a singleton (ex https://medium.com/swlh/node-js-and-singleton-pattern-7b08d11c726a) before resorting to fs since it's perhaps not important to persist across multiple command runs.